### PR TITLE
When an error occured in a JobQueue print it directly

### LINKED ISSUE #3536: Print build error while other jobs finish
When something goes wrong, Cargo prints
```
Build failed, waiting for other jobs to finish...
```
Then it does wait until everything finishes, and only then it prints the error message that it already knew about during that time. Meanwhile, I’m staring at it instead of starting to fix the issue since I don’t know yet what the issue is.

Could Cargo print the output of a failed job without waiting for other jobs to finish? (But still wait for those before exiting itself.)

### PR DESCRIPTION
**Pull request by dgellow**
Fix #3536

### CODE REVIEW

File: src/cargo/ops/cargo_rustc/job_queue.rs [hunk 1]
```diff
@@ -181,13 +181,14 @@ impl<'a> JobQueue<'a> {
                     match result {
                         Ok(()) => self.finish(key, cx)?,
                         Err(e) => {
+                            cx.config.shell().error(format!("{}", e))?;
```
Comment (alexcrichton) [inline src/cargo/ops/cargo_rustc/job_queue.rs]: Could this use Cargo's top level error printing functionality? That'll be sure to print the whole stack trace of the error. Basically something along the lines of [calling these functions](https://github.com/rust-lang/cargo/blob/cf4986dd4690591aead3edc9b42275430f7640c4/src/cargo/lib.rs#L153-L202) (but perhaps not exactly).

Comment (dgellow) [inline src/cargo/ops/cargo_rustc/job_queue.rs]: Yep, I think you're right, that's better.
I've made some modifications in `cargo/lib.rs` to handle that.

The result is like this:
- First call is a normal `cargo build`
- Second call has the flag `--verbosity`

screen shot 2017-01-25 at 19 59 01

**Comment by alexcrichton** [comment]
Nice! The new output looks great to me. Looks like there's some failures on Travis though?

**Comment by alexcrichton** [comment]
Looks like there's still one more failure perhaps?

**Comment by dgellow** [comment]
@alex done :)

**Comment by alexcrichton** [comment]
@bors: r+

Looks great, thanks!

**Pull request merged**